### PR TITLE
Set up Devise

### PR DESCRIPTION
**Pull request by martinkarugaba**
# :recycle: Changes made

- [ ] Installed devise in my blog app
- [ ] Added all the necessary methods

### CODE REVIEW

File: app/views/layouts/application.html.erb [hunk 1]
```diff
@@ -13,6 +13,8 @@
 
   <body>
     <main class="container mx-auto mt-28 px-5 flex flex-col justify-start items-center">
+      <p class="notice"><%= notice %></p>
+      <p class="alert"><%= alert %></p>
```
Comment (Abenezer-Tilahun) [inline app/views/layouts/application.html.erb:17]: - [ ] When I run the application I got the following error as shown with the screenshot related with **The asset "tailwind.css" is not present in the asset pipeline.** on these line `<%= stylesheet_link_tag "tailwind", "inter-font", "data-turbo-track": "reload" %>` So to run the app properly update the code line with `<%= stylesheet_link_tag "application.tailwind", "inter-font", "data-turbo-track": "reload" %>`.
- The error showed up when I run the app
image

File: app/views/devise/registrations/new.html.erb [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    <%= f.label :password_confirmation %><br />
+    <%= f.password_field :password_confirmation, autocomplete: "new-password" %>
+  </div>
+
+  <div class="actions">
+    <%= f.submit "Sign up" %>
+  </div>
+<% end %>
+
+<%= render "devise/shared/links" %>
```
Comment (Abenezer-Tilahun) [inline app/views/devise/registrations/new.html.erb]: - [ ] Good Job for setting up devise to  register a new user. But when I try to register with the **email, password and password confirmation** it throws an error related with Name can't be blanked  But for the user register there is no Name input that allow the user to insert the name So kindly adjust it.
- The signup page for register the user
image

- The Error message showed up after I input the email , password with the password confirmation
image

**Review by Abenezer-Tilahun — CHANGES_REQUESTED** [review]
Hi @martinkarugaba @richie1988 

Great job for implemented  the some of the Project Requirement  , but there are some changes that need to be implemented in order to proceed to the next project 👍

## Highlights 👏🏻

## Highlights 👏🏻

✔️ No linter errors.
✔️ You used correct Git-Flow.
✔️ Professional README  
✔️ Good Commit Messages.
✔️ Correctly Install and setup devise .
✔️ Good job for Implemented Blog app: add devise


almost

## Required Changes ♻️
_Check the comments below the review._

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.


**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by mukhammadrizooff — APPROVED** [review]
Hi @martinkarugaba, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉

 

## Highlights
- No linter errors ✔️ 
- Correct GitFlow ✔️ 
- Good implementations ✔️
- Descriptive `README` file ✔️

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**